### PR TITLE
Fix leaks

### PR DESCRIPTION
**Pull request by awlauria**
There's more to come, only 3 tracked down as of now.

### CODE REVIEW

File: src/server/pmix_server_ops.c [hunk 1]
```diff
@@ -4771,6 +4771,14 @@ static void tdes(pmix_server_trkr_t *t)
     if (NULL != t->info) {
         PMIX_INFO_FREE(t->info, t->ninfo);
     }
+    pmix_nspace_caddy_t *nm, *nm_next;
```
Comment (rhc54) [inline src/server/pmix_server_ops.c:4774]: Just wondering - would it make more sense to simply replace `PMIX_DESTRUCT(&t->nslist)` with `PMIX_LIST_DESTRUCT(&t->nslist)` here, and then add `PMIX_RELEASE(p->jobbkt)` to the `pmix_nspace_caddy_t` destructor on line 154 of src/include/pmix_globals.c? Seems to me like we always want to have these things removed/destructed.

Comment (awlauria) [reply src/server/pmix_server_ops.c:4774]: I like it, done.

**Comment by rhc54** [comment]
The Mellanox failure is odd as that change passed before it was committed - it has nothing to do with this one. I'll try to replicate and see what is going on.

File: src/include/pmix_globals.c [hunk 2]
```diff
@@ -154,6 +154,7 @@ static void ncdcon(pmix_nspace_caddy_t *p)
 static void ncddes(pmix_nspace_caddy_t *p)
 {
     if (NULL != p->ns) {
+        PMIX_RELEASE(p->ns->jobbkt);
```
Comment (rhc54) [inline src/include/pmix_globals.c:157]: I'm not sure, but you may need to check for `p->ns->jobbkt` being `NULL` prior to trying to release it - probably safer to do so.

**Review by rhc54 — APPROVED** [review]

**Pull request merged**